packages/wdio-junit-reporter/src/index.ts

```diff
@@ -126,11 +126,11 @@ class JunitReporter extends WDIOReporter {
                             }
                         } else {
                             // default
-                            testCase.error(step.error.message)
+                            testCase.failure(step.error.message)
                         }
                         testCase.standardError(`\n${step.error.stack}\n`)
                     } else {
-                        testCase.error()
+                        testCase.failure()
                     }
                     testCase.failure()
```

Comment:
I think with this change we would report a failure twice.

Reply:
I have removed the extra call. Anyway I have checked the [code](https://github.com/davidparsson/junit-report-builder/blob/main/src/test_case.js#L49-L58), it won't report the failure twice.

All tests are passing, but there is some issue with code coverage. This issue is evident with other closed pr's as well - `TypeError: Cannot read properties of undefined (reading 'endCol')`

Reply:
It's an issue with a dependency update we recently made. I am working on getting this fixed, so we can update your branch, which then should pass

Reply:
After merging the main branch, the unit tests are passing. But [e2e mac tests](https://github.com/webdriverio/webdriverio/actions/runs/8761339852/job/24047716176?pr=12715) are failing. Is there anything we can do to help?